packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/IJsiiObject.java

```diff
@@ -0,0 +1,59 @@
+package software.amazon.jsii;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a JavaScript object in the Java world.
+ */
+@Internal
+public interface IJsiiObject extends JsiiSerializable {
+    /**
+     * Calls a JavaScript method on the object.
+     * @param method The name of the method.
+     * @param nativeType The return type.
+     * @param args Method arguments.
+     * @param <T> Java type for the return value.
+     *
+     * @return A return value.
+     */
+    @Nullable
+    <T> T $jsii$call(final String method, final NativeType<T> nativeType, @Nullable final Object... args);
```

Comment:
Are people going to see this in their IDEs?

Reply:
Should these be `@Internal`?

Reply:
The interface is `@Internal` as a whole, so all members are implicitly `@Internal`.

This might show up in IDE's, yeah... This isn't *entirely* unheard of, but at least this gives a pretty clear signal that though shalt not touch this.

Reply:
Well if this is going to show up in IDE's, on  *all objects*, I'm less of a fan of putting the default implementation on the interface.

We only really need `$jsii$objectHandle()` or whatever you want to call it from the state of the interface. The implementation of `$jsii$call` could just be a static somewhere, so at least we don't pollute the autocomplete of every single object out there (too much).

Reply:
I've even inaugurated an external instance-to-objectID map... Would be nice if this could be "the standard", but I run into problems as this would require to be able to call `.hashCode` on instances *before* they are initialized, which isn't... something one should do.

But "pure" instances can now be used without any kind of inheritance from `JsiiObject`, which is a nice bonus, I think.